### PR TITLE
compare test config as objects instead of strings

### PR DESCRIPTION
**Pull request by apolcyn**
This also should have been made on v1.0.x, the PR was merged on master (https://github.com/grpc/grpc/pull/7576), but should be here too.

### CODE REVIEW

**Comment by dgquintas** [comment]
LGTM

**Comment by apolcyn** [comment]
failures look unrelated to csharp change here.

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
csharp basic tests are passing. merging this

File: src/csharp/Grpc.Core.Tests/SanityTest.cs [hunk 1]
```diff
@@ -58,10 +58,11 @@ public class SanityTest
         [Test]
         public void TestsJsonUpToDate()
         {
-            var discoveredTests = DiscoverAllTestClasses();
-            string discoveredTestsJson = JsonConvert.SerializeObject(discoveredTests, Formatting.Indented);
+            Dictionary<string, List<string>> discoveredTests = DiscoverAllTestClasses();
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Tests/SanityTest.cs]: nit: you could have used "var discoveredTests". Also below.

File: src/csharp/Grpc.Core.Tests/SanityTest.cs [hunk 2]
```diff
@@ -58,10 +58,11 @@ public class SanityTest
...
+			Dictionary<string, List<string>> testsFromFile
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Tests/SanityTest.cs]: indentation is broken here. Perhaps tabs have been used instead of spaces?

**Pull request merged**